### ISSUE TITLE
Some systems failed to build for Quicklisp dist

### ISSUE THREAD

**Issue by quicklisp** [Shinmera/for#6]
Building with SBCL 2.1.7.62-417920f09 / ASDF 3.3.5 for quicklisp dist creation.

Trying to build commit id 2e4fcfa0f9c1d2f4559c58cef20ccefa50ba180d

*for* fails to build with the following error:

```
; caught ERROR:
;   (during macroexpansion of (DEFINE-VALUE-BINDING ACROSS ...))
;   0 is not a string designator.
...
; caught ERROR:
;   COMMON-LISP:T names a defined constant, and cannot be used in LET.
...
; caught ERROR:
;   NIL cannot be used as a local variable.
...
Unhandled UIOP/LISP-BUILD:COMPILE-FILE-ERROR in thread #<SB-THREAD:THREAD "main thread" RUNNING {1001BF0103}>: COMPILE-FILE-ERROR while compiling #<CL-SOURCE-FILE "for" "standard">

```

[Full log here](http://report.quicklisp.org/2021-09-13/failure-report/for.html)

**Comment by xach** [comment]
Any ideas on this?

**Comment by Shinmera** [comment]
For what it's worth, I can't reproduce this with SBCL 2.1.8, so it might be a miscompilation.

**Comment by Shinmera** [comment]
Ah, nevermind, looks like lambda-fiddle's latest patch messes with things here.

**Comment by Shinmera** [comment]
Fixed in shinmera/lambda-fiddle#1b07e21
